Lib/test/test_imaplib.py

```diff
@@ -71,6 +71,15 @@ def test_that_Time2Internaldate_returns_a_result(self):
             imaplib.Time2Internaldate(t)
 
     def test_imap4_host_default_value(self):
+        # Check whether the IMAP4_PORT is truly unavailable.
+        with imaplib.socket.socket() as s:
```

Comment:
Hum wait, no, you cannot get socket from imaplib. You have to import is explicitly. imaplib.socket can disappear anytime and it's not part of it's public API. The test must not rely on implementation details.

Reply:
oh well, write hundred times "You shall not be clever."